### PR TITLE
Cleanup numeric interface

### PR DESCRIPTION
**Pull request by rijkvanzanten**
*  [x] Move inline input html to separate file
*  [x] Remove dependency on underscore
*  [x] Use [browser validation](http://caniuse.com/#search=validation) api

### CODE REVIEW

File: app/core/interfaces/numeric/index.js [hunk 1]
```diff
@@ -46,18 +32,21 @@ define([
           value = this.options.schema.get('default_value');
       }
 
+      var step = 1;
+
+      if (this.options.schema.get('type') === 'FLOAT' || this.options.schema.get('type') === 'VARCHAR') {
```
Comment (WellingGuzman) [inline app/core/interfaces/numeric/index.js]: Will `CHAR` allow steps?

File: app/core/interfaces/numeric/index.js [hunk 2]
```diff
@@ -68,31 +57,32 @@ define([
...
+      'DOUBLE',
       'VARCHAR',
       'CHAR',
-      'DOUBLE',
       'BIGINT'
     ],
     variables: [
       {id: 'size', type: 'String', default_value: 'large', ui: 'select', options: {options: {'large':__t('size_large'),'medium':__t('size_medium'),'small':__t('size_small')} }},
-      {id: 'placeholder_text', type: 'String', default_value: '', ui: 'textinput', char_length:200},
-      {id: 'allow_null', type: 'Boolean', default_value: false, ui: 'checkbox'},
```
Comment (WellingGuzman) [inline app/core/interfaces/numeric/index.js]: @rijkvanzanten Why this options was removed?

File: app/core/interfaces/numeric/index.js (hunk 1 above)
Comment (rijkvanzanten) [inline app/core/interfaces/numeric/index.js]: CHAR (and VARCHAR) will be saved as string and can therefor accept any step. I've added CHAR to the check as well in commit d176fe3d26c78e8bbbbcd8cd764e36f8816c9849

File: app/core/interfaces/numeric/index.js (hunk 2 above)
Comment (rijkvanzanten) [inline app/core/interfaces/numeric/index.js]: `allow_null` wasn't checked in any way and I thought the check would be the same as the one for `is_required`, please let me know if I overlooked something

**Pull request merged**